bpf/process/bpf_execve_event.c

```diff
@@ -67,8 +67,26 @@ event_args_builder(struct msg_execve_event *event)
 			return;
 
 		start_stack += off;
-		probe_arg_read(c, (char *)p, (char *)start_stack,
-			       (char *)end_stack);
+
+		if ((end_stack - start_stack) < BUFFER) {
+			probe_arg_read(c, (char *)p, (char *)start_stack,
+				       (char *)end_stack);
+		} else {
+			char *args = (char *)p + p->size;
+			__u32 size;
+
+			if (args >= (char *)&event->process + BUFFER)
+				return;
+
+			size = data_event_bytes(ctx, args,
```

Comment:
How about we use `(struct data_event_desc *)args` here and change `data_event_bytes` to:

`data_event_bytes(void *ctx, void *struct data_event_desc, ...)`

I think it would make the code somewhat clearer.

Reply:
I'll check, you might be right.. hopefully clang won't protest ;-)